hardhat-tests/test/internal/hardhat-network/stack-traces/test.ts

```diff
@@ -280,9 +281,15 @@ function compareStackTraces(
 
   // if IR is enabled, we ignore callstack entries in the comparison
   if (isViaIR) {
-    trace = trace.filter(
-      (frame) => frame.type !== StackTraceEntryType.CALLSTACK_ENTRY
```

Comment:
From my experience during this rewrite it was often the case that when the same type was removed and re-exported under the same name it was not recognized at run-time, even if the types were correct (meaning `tsc` was happy).

This was true for functions, classes and enums. What often helped was importing it under a different alias and then re-exporting it/using it with the same alias, removing the old code and then replacing with the old/actual name. Whether that's a problem with build cache or something I do not know (I tried cleaning node_modules, doing `pnpm install` etc. to no avail).

Rather than figuring out the underlying cause I decided not to do any further detective work and just apply the blunt workaround for now. Let's see if this is fixed with a subsequent change; I'd rather not spend more time here and focus on completing the porting work, instead.

Reply:
I didn't look into this, but @Xanewok did a very thorough research and couldn't find a good solution. [A follow-up PR](https://github.com/NomicFoundation/edr/pull/594) uses a different workaround, but it's still a workaround.

I don't think this is super important though, since it only affects tests' code.